packages/dev/loaders/src/USD/usdFileLoader.ts

```diff
@@ -0,0 +1,242 @@
+     * @param onProgress callback called while file is loading
+     * @param fileName Defines the name of the file to load
+     * @returns a promise containing the loaded meshes, particles, skeletons and animations
+     */
+    public async importMeshAsync(
+        meshesNames: any,
+        scene: Scene,
+        data: any,
+        rootUrl: string,
+        onProgress?: (event: ISceneLoaderProgressEvent) => void,
+        fileName?: string
+    ): Promise<ISceneLoaderAsyncResult> {
+        return this._parse(meshesNames, scene, data, rootUrl).then((meshes) => {
+            return {
+                meshes: meshes,
+                particleSystems: [],
+                skeletons: [],
+                animationGroups: [],
+                transformNodes: [],
+                geometries: [],
+                lights: [],
+                spriteManagers: [],
+            };
+        });
+    }
+    private static _UniqueResolveID = 1000;
+
+    private static _LoadScriptModuleAsync(scriptUrl: string, scriptId?: string): Promise<any> {
+        return new Promise((resolve, reject) => {
+            // Need a relay
+            let windowAsAny: any;
+            let windowString: string;
+
+            if (IsWindowObjectExist()) {
+                windowAsAny = window;
+                windowString = "window";
+            } else if (typeof self !== "undefined") {
+                windowAsAny = self;
+                windowString = "self";
+            } else {
+                reject(new Error("Cannot load script module outside of a window or a worker"));
+                return;
+            }
+
+            if (!windowAsAny._LoadScriptModuleResolve) {
+                windowAsAny._LoadScriptModuleResolve = {};
+            }
+            windowAsAny._LoadScriptModuleResolve[USDFileLoader._UniqueResolveID] = resolve;
+
+            scriptUrl += `
+                ${windowString}._LoadScriptModuleResolve[${USDFileLoader._UniqueResolveID}](returnedValue);
+                ${windowString}._LoadScriptModuleResolve[${USDFileLoader._UniqueResolveID}] = undefined;
+            `;
+            USDFileLoader._UniqueResolveID++;
+
+            Tools.LoadScript(
+                scriptUrl,
+                undefined,
+                (message, exception) => {
+                    reject(exception || new Error(message));
+                },
+                scriptId,
+                true
+            );
+        });
+    }
+
+    private _initializeTinyUSDZAsync(): Promise<void> {
+        return USDFileLoader._LoadScriptModuleAsync(
+            `
+            import Module from '${this._loadingOptions.usdLoaderUrl}/tinyusdz.js';
+            const returnedValue = await Module();
+            `
+        );
+    }
+
+    private _parse(meshesNames: any, scene: Scene, data: any, rootUrl: string): Promise<Array<AbstractMesh>> {
+        const babylonMeshesArray: Array<Mesh> = []; //The mesh for babylon
+
+        return new Promise((resolve) => {

```

Comment:
not needed, if the first call is a promise. could just be:

```javascript
return this._initializeTinyUSDZAsync().then((tinyusdzModule: any) => {
.....
   return babylonMeshesArray;
});
```

Reply:
Fixed